Fix/config set name

DimCitus opened the pull request:
Fix pg_autoctl config set pg_autoctl.name or hostname.

We need to take the new value in the local configuration file (that was ok)
and also update the monitor with the new value (that was not done).

Note that because of a change in the API where we need the whole keeper, the
function keeper_config_accept_new must move to the keeper.c file ("module").

onderkalaci reviewed (approved):
The actual code change is pretty small although github shows it as a large change.

Manually tested, all  looks good.  It'd be good to have a test for at least `set pg_autoctl.name`. I  guess  adding the test for  host could be complicated?

DimCitus commented on src/bin/pg_autoctl/keeper.c:
```diff
@@ -1704,7 +1704,7 @@ keeper_set_node_metadata(Keeper *keeper, KeeperConfig *oldConfig)
 		oldConfig->pgSetup.pgport == config->pgSetup.pgport)
 	{
-		log_trace("keeper_set_node_metadata: no changes");
+		log_warn("keeper_set_node_metadata: no changes");
```
This looks like a debug thing. That's what I do sometimes when I want to see some specific messages but I don't want to see all the TRACEs. Going to revert that change, thanks for noticing!

DimCitus replied:
Done now, thanks!

DimCitus commented:
> The actual code change is pretty small although github shows it as a large change.
> 
> Manually tested, all looks good. It'd be good to have a test for at least `set pg_autoctl.name`. I guess adding the test for host could be complicated?

I have added such a test. It requires waiting for a while, until the SIGHUP signal has been received by the running process, and then processed on the monitor. I have added a 2s sleep which seems to do the trick.

To be able to test a change of hostname we would have to be able to use another hostname that leads to the same node, or to handle transient failures to reach the node from the monitor. Another hostname would be either a DNS entry, and we don't have that, or another IP address, and we have a single IP address per network namespace in which we run our tests.

So yeah I agree with you, it looks too complex to organise testing around hostname changes.
